src/mscorlib/src/System/Span.cs

```diff
@@ -625,5 +636,266 @@ internal static unsafe void CopyTo<T>(ref T destination, ref T source, int eleme
                 }
             }
         }
+
+        internal static unsafe void ClearWithoutReferences(ref byte b, nuint byteLength)
+        {
+            if (byteLength == 0)
+                return;
+
+            fixed (byte* bytePointer = &b)
+            {
+                // Note: It's important that this switch handles lengths at least up to 22.
+                // See notes below near the main loop for why.
+
+                // The switch will be very fast since it can be implemented using a jump
+                // table in assembly. See http://stackoverflow.com/a/449297/4077294 for more info.
+
+                switch (byteLength)
```

Comment:
There is overhead in calling RhZeroMemory.

If you notice, for small span sizes (< 100 bytes), it is MUCH faster (almost 2-3x) not to call RhZeroMemory. For span sizes > 1k bytes, the current API is on par with calling RhZeroMemory unconditionally. For between 100-1k bytes, yes, calling RhZeroMemory unconditionally is slightly faster (but no more than 1.5x).

![image](https://cloud.githubusercontent.com/assets/6527137/22960671/902db6d4-f2f4-11e6-816a-22b84b1a6d7d.png)